core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -7706,8 +7704,7 @@ private static BiFunction<String, PropertyKey, PropertyKey> createNestedProperty
     private final PropertyType mType;
     private final Optional<Class<? extends Enum>> mEnumType;
     private final Optional<String> mDelimiter;
-    private BiFunction<String, PropertyKey, PropertyKey> mPropertyCreator =
-        PropertyCreators.DEFAULT_PROPERTY_CREATOR;
+    private BiFunction<String, PropertyKey, PropertyKey> mPropertyCreator;
```

Comment:
This should be final.

Reply:
Thanks for suggestion. If we make the final here, we will face the double claiming for mPropertyCreator. I would like to make this work. Then I will submit another pr to refactor the template constrictor to make it respect the best practice you mentioned.